tripper/convert/convert.py

```diff
@@ -35,115 +46,150 @@
     "requires": DCTERMS.requires,
     "label": RDFS.label,
     "comment": RDFS.comment,
+    "mapsTo": MAP.mapsTo,
 }
 
 
-def from_dict(
-    dct: "Mapping[str, Any]",
+def from_container(
+    container: "Union[Mapping[str, Any], Sequence[Any]]",
     iri: str,
-    bases: "Sequence" = (OTEIO.Dictionary,),
     lang: str = "en",
     recognised_keys: "Optional[Union[Dict, str]]" = None,
     keep: bool = False,
 ) -> list:
-    """Serialise a dict as RDF.
+    """Serialise a basic Python container type (mapping or sequence) as RDF.
 
     Arguments:
-        dct: The dict to be saved.
-        iri: IRI of indicidual that stands for the dict.
-        bases: Parent class(es) or the dict.
-        lang: Language to use for keys.
-        recognised_keys: An optional dict that maps dict keys that
+        container: The container to be saved.  Should be a mapping or
+            sequence.  The `load_container()` function will deserialise
+            them as dict and list, respectively.
+        iri: IRI of individual that stands for the container.
+        lang: Language to use for mapping keys.
+        recognised_keys: An optional dict that maps mapping keys that
             correspond to IRIs of recognised RDF properties.
             If set to the special string "basic", the
             `BASIC_RECOGNISED_KEYS` module will be used.
         keep: Whether to keep the key-value pair representation for
-            items serialised with recognised_keys.  Note that this
-            will duplicate potential large literal values.
+            mapping items serialised with recognised_keys.  Note that this
+            may duplicate potential large literal values.
 
     Returns:
         List of RDF triples.
+
+    Note:
+        `container` should not be an empty sequence.  The reason for this
+        is that is represented with rdf:nil, which is a single IRI and not
+        a triple.
     """
     if recognised_keys == "basic":
         recognised_keys = BASIC_RECOGNISED_KEYS
 
     rdf = []
-    for base in bases:
-        rdf.append((iri, RDF.type, base))
-
-    for dkey, dvalue in dct.items():
-        uuid = uuid4()
-
-        recognised = recognised_keys and dkey in recognised_keys
-
-        if isinstance(dvalue, Mapping):
-            # Ideally this should be a blank node, but that becomes
-            # too nested for rdflib.  Instead we make the IRI unique
-            # by embedding the UUID.
-            value = f"dict_{uuid}"
-            rdf.extend(
-                from_dict(
-                    dvalue,
-                    value,
-                    lang=lang,
-                    recognised_keys=recognised_keys,
-                    keep=keep,
-                )
-            )
-        elif dvalue is None:
-            value = OWL.Nothing
+
+    def get_obj_iri(obj, uuid):
+        """Return IRI for Python object `obj`.  The `uuid` argument is
```

Comment:
Please explain why this method returns either an IRI or a list

Reply:
The nested help-function get_obj_iri() will only return an IRI. I updated its docstring to explain that it also populates the list of triples (built up in the outer function from_container()) with a representation of `obj` .

`from_container()` will always return a list of triples.